subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -3323,11 +3336,17 @@ static void mfy_aux_offset_get(void *param)
 					   data_chan_map, data_chan_count);
 
 	/* Assertion check for delayed aux_offset calculations */
+	uint32_t ticks_to_start;
+	uint32_t ticks_elapsed;
+	uint32_t ticks_now;
+
 	ticks_now = ticker_ticks_now_get();
-	ticks_elapsed = ticker_ticks_diff_get(ticks_now, ticks_current);
-	ticks_to_start = HAL_TICKER_US_TO_TICKS(EVENT_OVERHEAD_XTAL_US) -
-			 HAL_TICKER_US_TO_TICKS(EVENT_OVERHEAD_PREEMPT_MIN_US);
-	LL_ASSERT_ERR(ticks_elapsed < ticks_to_start);
+	ticks_elapsed = ticker_ticks_diff_get(ticks_now, adv->ticks_at_expire);
+	ticks_to_start = HAL_TICKER_US_TO_TICKS(EVENT_OVERHEAD_XTAL_US);
+	ticks_to_start += HAL_TICKER_US_TO_TICKS(EVENT_OVERHEAD_START_US);
+	LL_ASSERT_MSG((ticks_elapsed <= ticks_to_start), "%s: elapsed = %u (%u) us.",
+		      __func__, HAL_TICKER_TICKS_TO_US(ticks_elapsed),
```

Comment:
Isn't the line number printed as part of the assert? if so, adding the function name seems superfluous

Reply:
I see. I guess we could consider handling this in the same way we do for LOG statements, where the function name is optionally printed, but that's of course more generic that just adding it a single place like here